dependencies.sh

```diff
@@ -27,10 +145,83 @@ else
 fi
 
 
+if cat /etc/os-release | grep "ubuntu" >> /dev/null; then
+
+        apt-get update
+
+        # Check if gcc is installed; if not, install it
+        if ! [ -x "$(command -v gcc)" ]; then
+        apt-get install -y gcc
+        fi
+
+        # Check if g++ is installed; if not, install it
+        if ! [ -x "$(command -v g++)" ]; then
+        apt-get install -y g++
+        fi
+
+        # Check if make is installed; if not, install it
+        if ! [ -x "$(command -v make)" ]; then
+        apt-get install -y make
+        fi
+
+elif cat /etc/os-release | grep -e "centos" >> /dev/null; then
+
+        yum update -y
+
+        # Check if gcc is installed; if not, install it
+        if ! [ -x "$(command -v gcc)" ]; then
+        yum install -y gcc
+        fi
+
+        # Check if g++ is installed; if not, install it
+        if ! [ -x "$(command -v g++)" ]; then
+        yum install -y gcc-c++
+        fi
+
+        # Check if make is installed; if not, install it
+        if ! [ -x "$(command -v make)" ]; then
+        yum install -y make
+        fi
+
+fi
+
+# install miniconda3
+if ! [ -x /usr/bin/miniconda3 ]
+then
+      wget https://repo.anaconda.com/miniconda/Miniconda3-py37_4.12.0-Linux-x86_64.sh \
```

Comment:
update the miniconda3 version

Reply:
Do you want the absolute latest version of miniconda to be installed (that requires python 3.9), or the latest version of miniconda that supports python 3.7?

Reply:
It's always 3.7
I am asking you to upgrade miniconda version. 4.23.xx is present and pip3 is not getting installed with earlier versions after they upgraded the conda version.